src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -386,9 +392,49 @@ public void StartProfileOptimization(string profile)
             InternalStartProfile(profile, m_pNativeAssemblyLoadContext);
 #endif // FEATURE_MULTICOREJI
         }
-        
+
+        public event EventHandler Unloading
+        {
+            [SecuritySafeCritical]
+            add
+            {
+                if (value == null)
```

Comment:
None of this should be required. The auto-generated C# code takes care of the thread safety.

The implementation of plumbing for this event should look just like the implementation for the `Resolving` event.